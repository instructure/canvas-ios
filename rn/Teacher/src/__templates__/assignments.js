//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: '1376',
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
  anonymize_students: false,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment({ assignment_group_id: '1' })],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
