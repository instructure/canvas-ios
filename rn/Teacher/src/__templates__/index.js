//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
