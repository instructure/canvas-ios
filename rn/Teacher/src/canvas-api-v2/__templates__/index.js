// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
