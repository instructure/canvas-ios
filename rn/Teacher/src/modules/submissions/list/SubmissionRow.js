//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  View,
  Image,
  TouchableHighlight,
} from 'react-native'
import DisclosureIndicator from '../../../common/components/DisclosureIndicator'
import Token from '../../../common/components/Token'
import i18n from 'format-message'
import { colors, createStyleSheet } from '../../../common/stylesheet'
import { Text } from '../../../common/text'
import SubmissionStatusLabel from './SubmissionStatusLabel'
import Avatar from '../../../common/components/Avatar'
import { formatGradeText, personDisplayName } from '../../../common/formatters'
import images from '../../../images'

type RowProps = {
  testID: string,
  onPress: () => void,
  children?: Array<any>,
  disclosure?: boolean,
}

export type SubmissionRowDataProps = {
  groupID?: string,
  user: {
    id: string,
    avatarUrl: string,
    name: string,
  },
  gradingType: GradingType,
  disclosure?: boolean,
  disabled?: boolean,
  submission: Object,
}

export type SubmissionRowProps = {
  onPress: (userID: string) => any,
  onAvatarPress?: Function,
  anonymous: boolean,
} & SubmissionRowDataProps

class Row extends Component<RowProps, any> {
  render () {
    const { onPress, testID, children, disclosure } = this.props
    return (
      <View>
        <TouchableHighlight style={styles.touchableHighlight} onPress={onPress} testID={testID} accessibilityTraits={['button']}>
          <View style={styles.container}>
            {children}
            {disclosure && <DisclosureIndicator />}
          </View>
        </TouchableHighlight>
      </View>
    )
  }
}

export const Grade = ({ submission, gradingType }) => {
  let gradeText = submission.grade

  if (submission == null || submission.state === 'unsubmitted' || submission.gradingStatus === 'needs_grading') {
    gradeText = '--'
  } else if (submission.excused) {
    gradeText = i18n('Excused')
  } else {
    gradeText = formatGradeText({ grade: submission.grade, score: submission.score, gradingType })
  }

  return <Text style={[ styles.gradeText, { alignSelf: 'center' } ]}>{ gradeText }</Text>
}

class SubmissionRow extends Component<SubmissionRowProps, any> {
  onPress = () => {
    // $FlowFixMe
    this.props.onPress(this.props.user.id)
  }

  onAvatarPress = () => {
    if (this.props.onAvatarPress) {
      this.props.onAvatarPress(this.props.user.id)
    }
  }

  render () {
    let { user, group, submission, gradingType } = this.props
    let contextID = group?.id ?? user?.id
    let name = group?.name ?? personDisplayName(user?.name, user?.pronouns)
    let avatarName = group?.name ?? user?.name
    let avatarURL = user.avatarUrl

    if (this.props.anonymous) {
      let anonymousName = group != null ? i18n('Group') : i18n('Student')
      name = anonymousName
      avatarName = anonymousName
      avatarURL = null
    }

    return (
      <Row testID={`submission-${contextID}`} onPress={this.onPress}>
        <View style={styles.avatar}>
          <Avatar
            key={contextID}
            avatarURL={avatarURL}
            userName={avatarName}
            onPress={!this.props.anonymous && this.props.onAvatarPress && this.onAvatarPress}
          />
        </View>
        <View style={styles.textContainer}>
          <Text
            style={styles.title}
            ellipsizeMode='tail'
            numberOfLines={2}>{name}</Text>
          {gradingType !== 'not_graded' &&
            <SubmissionStatusLabel submission={submission} />
          }
          {submission.gradingStatus === 'needs_grading' &&
            <Token style={{ alignSelf: 'flex-start', marginTop: 8 }} color={ colors.textInfo }>
              {i18n('Needs Grading')}
            </Token>
          }
        </View>
        <Grade submission={submission} gradingType={gradingType} />
        {submission.grade != null && submission.postedAt == null &&
          <Image
            source={images.off}
            testID='SubmissionRow.hiddenIcon'
            style={{ tintColor: colors.textDanger, height: 20, width: 20, marginLeft: 10 }}
          />
        }
      </Row>
    )
  }
}

export default SubmissionRow

const styles = createStyleSheet(colors => ({
  touchableHighlight: {
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: colors.backgroundLightest,
    alignItems: 'center',
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 16,
    paddingRight: 16,
  },
  textContainer: {
    flex: 1,
    paddingLeft: 8,
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
    color: colors.textDarkest,
  },
  gradeText: {
    fontSize: 14,
    fontWeight: '600',
    color: colors.textDarkest,
  },
  avatar: {
    width: 40,
    height: 40,
    marginRight: 8,
  },
}))
