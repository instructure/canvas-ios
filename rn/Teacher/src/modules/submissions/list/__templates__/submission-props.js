//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
