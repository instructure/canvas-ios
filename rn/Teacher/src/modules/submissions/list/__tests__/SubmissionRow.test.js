//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import { shallow } from 'enzyme'
import 'react-native'
import React from 'react'
import SubmissionRow from '../SubmissionRow'
import explore from '../../../../../test/helpers/explore'
import renderer from 'react-test-renderer'

import * as templates from '../../../../canvas-api-v2/__templates__'

jest
  .mock('react-native/Libraries/Components/Touchable/TouchableHighlight', () => 'TouchableHighlight')
  .mock('../../../../common/components/Avatar', () => 'Avatar')

const defaultProps = {
  user: templates.user({
    id: '1',
    name: 'Green Latern',
    avatarUrl: 'https://cats.pajamas/',
  }),
  submissionID: null,
  submission: templates.submission(),
  anonymous: false,
  onPress: jest.fn(),
  onAvatarPress: jest.fn(),
  gradingType: 'points',
}

test('unsubmitted ungraded row renders correctly', () => {
  let submission = templates.submission({
    submittedAt: null,
    grade: null,
    state: 'unsubmitted',
  })
  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('missing ungraded row renders correctly', () => {
  let submission = templates.submission({
    state: 'unsubmitted',
    submittedAt: null,
    missing: true,
    grade: null,
  })
  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('late graded row renders correctly', () => {
  const submission = templates.submission({
    late: true,
    grade: 'B-',
  })
  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} gradingType='gpa_scale' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('submitted ungraded row renders correctly', () => {
  const submission = templates.submission({
    submittedAt: new Date().toISOString(),
    gradingStatus: 'needs_grading',
  })

  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('submitted not_graded row renders correctly', () => {
  const submission = templates.submission({
    submittedAt: new Date().toISOString(),
    gradingStatus: 'needs_grading',
  })

  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} gradingType='not_graded' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('excused row renders correctly', () => {
  const submission = templates.submission({
    missing: true,
    excused: true,
  })
  let tree = renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress called on tap', () => {
  const onPress = jest.fn()
  let submission = templates.submission()
  let row = explore(renderer.create(
    <SubmissionRow {...defaultProps} submission={submission} onPress={onPress} />
  ).toJSON()).selectByID(`submission-${defaultProps.user.id}`)
  row && row.props.onPress()
  expect(onPress).toHaveBeenCalledWith(defaultProps.user.id)
})

test('anonymous grading doesnt show users names', () => {
  let submission = templates.submission({
    state: 'unsubmitted',
    submittedAt: null,
    grade: null,
  })
  let tree = renderer.create(
    <SubmissionRow
      {...defaultProps}
      submission={submission}
      anonymous
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders name', () => {
  let user = templates.user({
    id: '1',
    name: 'Alice',
  })
  let tree = shallow(<SubmissionRow {...defaultProps} user={user} />)
  expect(tree.find('[children="Alice"]').exists()).toBe(true)
})

test('renders pronouns', () => {
  let user = templates.user({
    id: '1',
    name: 'Alice',
    pronouns: 'She/Her',
  })
  let tree = shallow(<SubmissionRow {...defaultProps} user={user} />)
  expect(tree.find('[children="Alice (She/Her)"]').exists()).toBe(true)
})

test('pressing the avatar calls onAvatarPress', () => {
  const onAvatarPress = jest.fn()
  let tree = renderer.create(
    <SubmissionRow {...defaultProps} onAvatarPress={onAvatarPress} />
  ).toJSON()
  let avatar = explore(tree).selectByType('Avatar')
  avatar.props.onPress()
  expect(onAvatarPress).toHaveBeenCalledWith('1')
})

test('anonymous avatar press', () => {
  let spy = jest.fn()
  let tree = shallow(<SubmissionRow {...defaultProps} anonymous={true} onAvatarPress={spy} />)
  tree.find('Avatar').simulate('Press')
  expect(spy).not.toHaveBeenCalled()
})

test('shows the eyeball when grades are not posted', () => {
  let submission = templates.submission({
    submittedAt: '2017-04-05T15:12:45Z',
    grade: '20%',
    postedAt: null,
  })
  let tree = renderer.create(
    <SubmissionRow
      {...defaultProps}
      submission={submission}
    />
  ).toJSON()
  let eye = explore(tree).selectByID('SubmissionRow.hiddenIcon')
  expect(eye).toBeTruthy()
})

test('does not show the eyeball when grade is posted', () => {
  let submission = templates.submission({
    submittedAt: '2017-04-05T15:12:45Z',
    grade: '20%',
    postedAt: '2019-08-29T00:00:00.000Z',
  })
  let tree = renderer.create(
    <SubmissionRow
      {...defaultProps}
      submission={submission}
    />
  ).toJSON()
  let eye = explore(tree).selectByID('SubmissionRow.hiddenIcon')
  expect(eye).toBeFalsy()
})

test('does not show the eyeball when not graded', () => {
  let submission = templates.submission({
    submittedAt: null,
    grade: null,
    postedAt: null,
  })

  let tree = renderer.create(
    <SubmissionRow
      {...defaultProps}
      submission={submission}
    />
  ).toJSON()
  let eye = explore(tree).selectByID('SubmissionRow.hiddenIcon')
  expect(eye).toBeFalsy()
})
