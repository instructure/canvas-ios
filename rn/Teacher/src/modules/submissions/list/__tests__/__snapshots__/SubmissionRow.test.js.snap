// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous grading doesnt show users names 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL={null}
          onPress={false}
          userName="Student"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Student
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#8B969E",
                },
              ],
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        --
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;

exports[`excused row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        Excused
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/off.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginLeft": 10,
            "tintColor": "#EE0612",
            "width": 20,
          }
        }
        testID="SubmissionRow.hiddenIcon"
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`late graded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#FC5E13",
                },
              ],
            ]
          }
        >
          Late
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        B-
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/off.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginLeft": 10,
            "tintColor": "#EE0612",
            "width": 20,
          }
        }
        testID="SubmissionRow.hiddenIcon"
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`missing ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#EE0612",
                },
              ],
            ]
          }
        >
          Missing
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        --
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;

exports[`submitted not_graded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          color="#008EE2"
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "fontSize": 11,
                  "fontWeight": "500",
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 3,
                  "textAlign": "center",
                },
                Object {
                  "borderColor": "#008EE2",
                  "color": "#008EE2",
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginTop": 8,
                },
              ],
            ]
          }
        >
          NEEDS GRADING
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        --
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/off.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginLeft": 10,
            "tintColor": "#EE0612",
            "width": 20,
          }
        }
        testID="SubmissionRow.hiddenIcon"
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`submitted ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#00AC18",
                },
              ],
            ]
          }
        >
          Submitted
        </Text>
        <Text
          color="#008EE2"
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "fontSize": 11,
                  "fontWeight": "500",
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 3,
                  "textAlign": "center",
                },
                Object {
                  "borderColor": "#008EE2",
                  "color": "#008EE2",
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginTop": 8,
                },
              ],
            ]
          }
        >
          NEEDS GRADING
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        --
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/off.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginLeft": 10,
            "tintColor": "#EE0612",
            "width": 20,
          }
        }
        testID="SubmissionRow.hiddenIcon"
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`unsubmitted ungraded row renders correctly 1`] = `
<View>
  <TouchableHighlight
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="submission-1"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Avatar
          avatarURL="https://cats.pajamas/"
          onPress={[Function]}
          userName="Green Latern"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 8,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Green Latern
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "paddingTop": 2,
                },
                undefined,
                Object {
                  "color": "#8B969E",
                },
              ],
            ]
          }
        >
          Not Submitted
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        --
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;
